exercises/07-normalizr/backend/server.ts

```diff
@@ -0,0 +1,77 @@
+import express from 'express'
+
+import { User, UserData } from '../src/modules/users/user-types'
+
+const port = 3001
+
+const areStringsSame = (a: string, b: string): boolean =>
+  a.toLowerCase() === b.toLowerCase()
+
+const hasSameName = (
+  user: UserData,
+  firstName: string,
+  lastName: string
+): boolean =>
+  areStringsSame(firstName, user.firstName) &&
+  areStringsSame(lastName, user.lastName)
+
+const findRegnalNumber = (users: User[], userName: UserData) => {
+  const regnalNumbers = users
+    .filter((user) => hasSameName(user, userName.firstName, userName.lastName))
+    .map(({ regnalNumber }) => regnalNumber)
+
+  if (regnalNumbers.length === 0) {
+    return 1
+  }
+
+  return Math.max(...regnalNumbers) + 1
+}
+
+const skills = [
+  {
+    id: 'skill-1',
+    name: 'Change my face',
+  },
+  {
+    id: 'skill-2',
+    name: 'Control dragons',
+  },
+]
+
+const users: User[] = []
+
+const app = express()
+app.use(express.urlencoded({ extended: true }))
+app.use(express.json())
+
+app.get('/users', (_req, res) => res.json(users))
```

Comment:
I'd replace _req with _ 🤔

Reply:
Good practice to wrap the handler into some error handler... (try/catch) just to make sure request won't hang in case of exception.

Reply:
This is not necessarily connected to this endpoint... but rather post/users (or generally everywhere with a middleware)

Reply:
Renamed to `_`, see https://github.com/salsita/react-training/pull/59/commits/c7b960b599747f88aebd2d95c3e13abc81385778

Reply:
@tomkis The `Express` has a built-in [error handling](https://expressjs.com/en/guide/error-handling.html):
> Errors that occur in synchronous code inside route handlers and middleware require no extra work. If synchronous code throws an error, then Express will catch and process it.

The default behavior would be among other things to set `statusCode` and `statusMessage` to an error in the response.

This could be enough for the training exercises.

Reply:
I've also added an input validation to `post` request in 8be4677aa6cc9c9067d61d6d2513cfe72551def3